File: day-10/python/part2.py
#!/usr/bin/env python3

import sys
from collections import defaultdict

adapter_list = sorted([0] + [int(x.strip()) for x in open(sys.argv[1], "r").readlines()])

highest = adapter_list[-1] + 3
adapter_list.append(highest)
adapter_range = list(range(1, 4))

combos = defaultdict(lambda: 0)
combos[0] = 1
    
for x in adapter_list:
    combos[x] += combos[x - 1] + combos[x - 2] + combos[x - 3]

print(combos[max(combos)])

File: day-10/nim/part2.nim
from os import paramStr
from algorithm import sorted
import strutils, sequtils, tables

var adapter_list = readFile(paramStr(1)).split.map(parseInt).sorted

var
  highest = adapter_list[^1] + 3
  combos = [0].toCountTable # bascially defaultdict(lambda: 0)

adapter_list.add(highest)
combos[0] = 1

for x in adapter_list:
  combos[x] = combos[x - 1] + combos[x - 2] + combos[x - 3]

echo combos[highest-3]
